yoyodyne/models/hard_attention.py

```diff
@@ -56,15 +60,25 @@ def __init__(
         self.classifier = nn.Linear(
             self.decoder.output_size, self.target_vocab_size
         )
-        assert self.teacher_forcing, """Teacher forcing flag disabled
-            but this model requires teacher forcing for training."""
+        assert (
+            self.teacher_forcing
+        ), "Teacher forcing disabled but required by this model"
 
     def init_decoding(
         self, encoder_out: torch.Tensor, encoder_mask: torch.Tensor
     ) -> Tuple[torch.Tensor, torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]:
         """Initializes hiddens and initial decoding output.
 
-        Simply feeds a BOS string to decoder to provide initial probability.
+        This feeds the BOS string to the decoder to provide an initial
+        probability.
+
+        Args:
+            encoder_out (torch.Tensor).
+            encoder_mask (torch.Tensor).
+
+        Returns:
+            Tuple[torch.Tensor, torch.Tensor, Tuple[torch.Tensor,
```

Comment:
Not a bad idea---I think @Adamits did this for module outputs---but if it's okay with you I'd like to put that in a different PR. Implementationally it may not be as simple as just making a named tuple if we need to keep it connected with the compute graph. (See [here](https://github.com/CUNY-CL/yoyodyne/blob/a96100f49f6c619329019a8f4083e68040d63a1a/yoyodyne/data/batches.py#L49) for an example of what one does if that's the case.)

Reply:
Creating minor issue to tag to PR.